bin/get_input_data.R

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env Rscript 
+
+# extract reference and query data to be used in the control workflow
+# required data include:
+# - reference expression matrix (10X dir)
+# - query expression matrix 
+# - reference SDRF file (need cell ids, inferred cell types, and CL terms)
+# - marker genes file for reference dataset 
+
+suppressPackageStartupMessages(require(optparse))
+suppressPackageStartupMessages(require(workflowscriptscommon))
+suppressPackageStartupMessages(require(R.utils))
+
+option_list = list(
+    make_option(
+        c("-a", "--reference-mat-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-b", "--reference-barcodes-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-c", "--reference-genes-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-d", "--query-matrix-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-e", "--query-barcodes-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-f", "--query-genes-url"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-g", "--reference-metadata"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    ),
+    make_option(
+        c("-i", "--marker-genes-file"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = ""
+    )
+)
+
+opt = wsc_parse_args(option_list, mandatory = c("reference_mat_url", "reference_barcodes_url", "reference_genes_url",
+                                                 "query_matrix_url", "query_barcodes_url", "query_genes_url",
+                                                 "reference_metadata", "marker_genes_file"))
+
+urls = unlist(opt[1:8])
+# standard file names
+names = c(rep(c("matrix.mtx", "barcodes.tsv",  "genes.tsv"), 2), "ref_metadata.txt", "ref_marker_genes.txt")
+# set up directories
+ref_dir = "reference_10X_dir/"
+query_dir = "query_10X_dir/"
+dir.create(ref_dir)
+dir.create(query_dir)
+
+for(idx in 1:length(names)){
+    if(idx <=3) d = paste0(ref_dir, names[idx], sep = "")
```

Comment:
Could you add code comments describing what this logic is for please?

Reply:
Actually, this logic depending on indices is rather convoluted- could you simplify? Maybe what is required in fact is a separate script that creates the 10X directory structure for a single dataset, with parameters? Then you could have separate workflow processes which retrieve the index and query sets. Plus you'd have a re-usable component. 

@pcm32 hacked something together for Galaxy: https://github.com/ebi-gene-expression-group/container-galaxy-sc-tertiary/blob/develop/tools/tertiary-analysis/data-scxa/retrieve-scxa.xml. Maybe we could have an analagous but tidier thing for the CLI layer.